engine/src/main/java/org/hibernate/search/v6poc/engine/impl/SearchMappingRepositoryImpl.java

```diff
@@ -15,24 +15,20 @@
 import org.hibernate.search.v6poc.util.SearchException;
 import org.hibernate.search.v6poc.util.spi.Closer;
 
-
-/**
- * @author Yoann Rodiere
- */
 public class SearchMappingRepositoryImpl implements SearchMappingRepository {
 
-	private final Map<MappingKey<?>, MappingImplementor> mappings;
+	private final Map<MappingKey<?>, MappingImplementor<?>> mappings;
 	private final Map<String, Backend<?>> backends;
 
-	SearchMappingRepositoryImpl(Map<MappingKey<?>, MappingImplementor> mappings, Map<String, Backend<?>> backends) {
+	SearchMappingRepositoryImpl(Map<MappingKey<?>, MappingImplementor<?>> mappings, Map<String, Backend<?>> backends) {
 		this.mappings = mappings;
 		this.backends = backends;
 	}
 
 	@Override
 	public <M> M getMapping(MappingKey<M> mappingKey) {
 		@SuppressWarnings("unchecked") // See SearchMappingRepositoryBuilderImpl: we are sure that, if there is a mapping, it implements M
-		M mapping = (M) mappings.get( mappingKey );
+		M mapping = (M) mappings.get( mappingKey ).toAPI();
```

Comment:
I don't see the purpose of this `toAPI()` method? I mean your objects are inheriting from the API type so why would you need this method?

Reply:
I don't want to expose the MappingImplementor interface through the APIs, but I need the mappings to implement it for the benefit of the SearchMappingRepository.
In particular I don't want users to start closing the mappings, because it's the responsibility of the SearchMappingRepository.

Reply:
To clarify: the type M here will be a user-facing interface, not the implementation type. So the user-facing will not extend MappingImplementor, but the implementation type will.